taqueria-plugin-tezos-client/client.ts

```diff
@@ -0,0 +1,256 @@
+            sendErr(" ")
+            sendErr(err.message.split("\n").slice(1).join("\n"))
+            return Promise.resolve({
+                contract: sourceFile,
+                result: "Invalid"
+            })
+        })
+    }
+
+    return execCmd(getCheckFileExistenceCommand(sandbox, sourcePath))
+    .then(typecheckContractHelper)
+    .catch(err => {
+        sendErr(" ")
+        sendErr(sourceFile + ": Does not exist\n")
+        return Promise.resolve({
+            contract: sourceFile,
+            result: "N/A"
+        })
+    })
+}
+
+const typecheckMultiple = (opts: Opts, sandbox: Sandbox) => (sourceFiles: string[]) : Promise<{contract: string, result: string}[]> => {
+    return Promise.all(sourceFiles.map(typecheckContract(opts, sandbox)))
+}
+
+const typecheckAll = (opts: Opts, sandbox: Sandbox): Promise<{contract: string, result: string}[]> => {
+    // TODO: Fetch list of files from SDK
+    return glob(['**/*.tz'], {cwd: opts.artifactsDir, absolute: false})
+    .then(entriesFromArtifactsDir =>
+        glob(['**/*.tz'], {cwd: opts.contractsDir, absolute: false})
+        .then(entriesFromContractsDir =>
+            [...new Set([...entriesFromArtifactsDir, ...entriesFromContractsDir])].map(typecheckContract(opts, sandbox))
+        )
+    )
+    .then(promises => Promise.all(promises))
+}
+
+const typecheck = <T>(parsedArgs: Opts, sandbox: Sandbox): LikeAPromise<PluginResponse, Failure<T>> => {
+    const sourceFiles = (parsedArgs.sourceFiles as string).split(',')
+    let p;
+    if (parsedArgs.sourceFiles) {
+        if (sourceFiles.length == 1)
+            p = typecheckContract (parsedArgs, sandbox) (parsedArgs.sourceFiles as string).then(data => [data])
+        else
+            p = typecheckMultiple (parsedArgs, sandbox) (sourceFiles)
+    } else {
+        p = typecheckAll (parsedArgs, sandbox)
+            .then(results => {
+                if (results.length === 0) sendErr("No contracts found to compile.")
+                return results
+            })
+    }
+    return p.then(sendAsyncJsonRes)
+}
+
+const typecheckTask = async <T>(parsedArgs: Opts) : LikeAPromise<void, Failure<T>> => {
+    const sandbox = getSandbox(parsedArgs)
+    if (sandbox) {
+        if (doesUseFlextesa(sandbox)) {
+            return await isSandboxRunning(sandbox.name.value)
+            ? typecheck(parsedArgs, sandbox).then(sendJsonRes)
+            : sendAsyncErr(`The ${sandbox.name} sandbox is not running.`)
+        }
+        return sendAsyncErr(`Cannot start ${sandbox.label} as its configured to use the ${sandbox.plugin} plugin.`)
+    }
+    return sendAsyncErr(`There is no sandbox configuration with the name ${parsedArgs.sandboxName}.`)
+}
+
+//////////// Simulate task ////////////
+
+const getStorageFromConfig = (opts: Opts, sourceFile: string) : string|undefined => {
+    const config = opts.config
+    const defaultEnv = config.environment.default
+    const storages = config.environment[defaultEnv].storage
```

Comment:
For now, it only supports the default environment. But it wouldn't be hard to extend.

Reply:
With the current tasks implemented, the only use case I can think of would be to run the `taq typecheck` task against 2 different protocols using the `-e` cli flag

But, since we can currently handle this right in the task with the `--sandbox` option/flag, it's not actually a product gap so you can decide whether to implement now or defer

Reply:
Since any sandbox will work for now, I will defer until we need it